src/Entity/CivicrmEntity.php

```diff
@@ -39,6 +61,7 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       $fields[$civicrm_field['name']] = $field_definition_provider->getBaseFieldDefinition($civicrm_field);
       $fields[$civicrm_field['name']]->setRequired(isset($civicrm_required_fields[$civicrm_field['name']]));
     }
+    $fields['drupal_crud'] = BaseFieldDefinition::create('boolean')->setComputed(TRUE);
```

Comment:
Question: Do we really need to declare a Drupal field for this? Or, could we declare just a plain class property on the `\Drupal\civicrm_entity\Entity\CivicrmEntity` class?

I don't know the answer to that question - it's a real question. :-) But it would seem nice to not have that be a Drupal field, but just a class property. Having it be a Drupal field means it could appear in some weird places, like grabbing the entity via the REST API.

Reply:
Just double-checked, without this the value doesn't persist throughout the save process and is not available in the Civi pre and post hooks.

I don't particularly like this either, open to other suggestions on how to make it work without defining as a field.

Reply:
Ok! My feeling is if there isn't an easy answer, this is probably fine for now, and we can always fix later if someone comes up with a better way.